packages/backend/src/services/spsp.ts

```diff
@@ -0,0 +1,91 @@
+import Axios from 'axios'
+import base64url from 'base64url'
+import * as uuid from 'uuid'
+import { Middleware } from 'koa'
+import { StreamServer } from '@interledger/stream-receiver'
+import { AppContainer, AppContext } from '../app'
+import { User } from '../models'
+
+export async function makeSPSPHandler(
+  container: AppContainer
+): Promise<Middleware<unknown, AppContext>> {
+  const config = await container.use('config')
+  const axios = Axios.create({
+    baseURL: config.accountsUrl,
+    timeout: 10_000,
+    validateStatus: (status) => status === 200 || status === 404
+  })
+  const server = new StreamServer({
+    serverSecret: config.streamSecret,
+    serverAddress: config.ilpAddress
+  })
+
+  return async function handleSPSP(ctx: AppContext): Promise<void> {
+    if (!uuid.validate(ctx.params.id) || uuid.version(ctx.params.id) !== 4) {
+      ctx.throw(400, 'Failed to generate credentials: invalid user id')
+    }
+
+    if (!ctx.get('accept').includes('application/spsp4+json')) {
```

Comment:
I see "application/spsp4+json" referenced in multiple spots.  Will this string ever change?  If so, do you think it should be moved to a constant somewhere?